### PR TITLE
Adds new temperature endpoint /tas2km to API which queries the 2km historical and projected temperature coverages

### PR DESCRIPTION
**Pull request by BobTorgerson**
This PR adds the new 2km temperature endpoint containing three models (5ModelAvg, GFDL-CM3, and NCAR-CCSM4), two scenarios (RCP 4.5 & RCP 8.5), all months, and all years for both historical (1901-2015) and projected (2006-2100). 

The endpoint is only for point data currently:
localhost:5000/tas2km/point/65/-155

This endpoint has replaced the previous EDS endpoint for temperature which can be seen in:
localhost:5000/eds/temperature/65/-155
localhost:5000/eds/all/65/-155

The coverages for the historical and projected data only exist on Apollo currently so be sure to set your env var appropriately:
export API_RAS_BASE_URL=https://apollo.snap.uaf.edu/rasdaman/

### CODE REVIEW

**Comment by BobTorgerson** [comment]
@charparr @cstephen I think this API update is ready for review!

File: routes/taspr.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import itertools
 from urllib.parse import quote
 import numpy as np
+import pandas as pd
+import json
```
Comment (cstephen) [inline routes/taspr.py:8]:
```suggestion
```

The json module is an unused import here.

File: templates/documentation/taspr.html [hunk 2]
```diff
@@ -405,6 +426,89 @@ <h4>Annual mean temperature or annual total precipitation</h4>
...
+          ...
+        },
+        ...
+      },
+      ...
+    },
+    ...
+  }
+}
+
```
Comment (cstephen) [inline templates/documentation/taspr.html:511]:
```suggestion
</pre>
```
Just need to close the `<pre>` block here.

File: templates/documentation/taspr.html [hunk 3]
```diff
@@ -154,6 +154,27 @@ <h4>Yearly values (point query)</h4>
 
 <h4>Yearly values within provided year range (point query)</h4>
```
Comment (cstephen) [inline templates/documentation/taspr.html:155]:
```suggestion
<h4>Monthly min, mean, and maximum temperature for each year (point query)</h4>
```
Update this `<h4>` heading to match the corresponding output section.

File: routes/taspr.py [hunk 4]
```diff
@@ -1188,6 +1493,48 @@ def mmm_point_data_endpoint(
...
+            ),
+            422,
+        )
+
+    # try:
+    point_pkg = asyncio.run(run_fetch_tas_2km_point_data(lat, lon))
+    # except Exception as exc:
+    #     if hasattr(exc, "status") and exc.status == 404:
+    #         return render_template("404/no_data.html"), 404
+    #     return render_template("500/server_error.html"), 500
```
Comment (cstephen) [inline routes/taspr.py:1525]:
```suggestion
    try:
        point_pkg = asyncio.run(run_fetch_tas_2km_point_data(lat, lon))
    except Exception as exc:
        if hasattr(exc, "status") and exc.status == 404:
            return render_template("404/no_data.html"), 404
        return render_template("500/server_error.html"), 500
```

File: routes/eds.py [hunk 5]
```diff
@@ -35,6 +35,22 @@ async def fetch_data(url):
...
 
+async def fetch_data_with_retry(url, max_retries=3):
+    for retry in range(max_retries):
+        response = await fetch_data(url)
+        # If the response is not blank, return the response
+        if response != dict():
+            return response
+        elif retry < max_retries - 1:
+            # Sleep for a moment before retrying the given endpoint
+            print(f"Retrying {url} after attempt {retry + 1}")
```
Comment (cstephen) [inline routes/eds.py:46]:
```suggestion
            app.logger.info(f"Retrying {url} after attempt {retry + 1}")
```

**Review by cstephen — CHANGES_REQUESTED** [review]
This new endpoint looks good and behaves as expected 👍 I found a handful of minor issues, all of which I've added with suggestions in the review comments.

**Review by charparr — APPROVED** [review]
Alright with Craig's suggestions and the most recent commit https://github.com/ua-snap/data-api/pull/357/commits/f569a704dda619012454cc80c1dc6330c76651e0 this is looking good! I've hitting the expected No Data / Invalid Latlon / Bad Request errors and also returning data as expected via routes like http://127.0.0.1:5000/tas2km/point/65.0628/-146.1627 and http://127.0.0.1:5000/eds/all/65/-145.

CSV output looks good for the `tas2km` routes too!

File: routes/eds.py (hunk 5 above)
Comment (cstephen) [reply routes/eds.py:46]: Bruce suggested using `app.logger.warn` instead.

**Pull request merged**